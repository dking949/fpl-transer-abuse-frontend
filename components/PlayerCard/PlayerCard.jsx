/* eslint-disable camelcase */
import React from 'react';
import CardOverflow from '@mui/joy/CardOverflow';
import Typography from '@mui/joy/Typography';
import AspectRatio from '@mui/joy/AspectRatio';
import Image from 'next/image';
import CardStyled from './styled';
import { Player } from '../../types';

function PlayerCard({ player_name, points, photo_url }) {
  return (
    <CardStyled variant="outlined">
      <CardOverflow>
        <AspectRatio objectFit="contain" maxHeight="300px">
          <Image
            src={photo_url}
            width={300}
            height={300}
            alt="Player Pic"
          />
        </AspectRatio>
      </CardOverflow>
      <Typography level="h2" sx={{ fontSize: 'md', mt: 2 }}>
        {player_name}
      </Typography>
      <Typography level="body2" sx={{ mt: 0.5, mb: 2 }}>
        Points:
        {' '}
        {points}
      </Typography>
    </CardStyled>
  );
}

PlayerCard.propTypes = Player;

export default PlayerCard;
