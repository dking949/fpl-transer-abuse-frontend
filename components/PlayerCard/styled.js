import Card from '@mui/joy/Card';
import styled from 'styled-components';

const CardStyled = styled(Card)`
  text-align: center;
  overflow: auto;
`;

export default CardStyled;
