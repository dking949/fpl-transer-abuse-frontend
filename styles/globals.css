html,
body {
  padding: 0;
  margin: 0;
  font-family: "Open Sans", "sans-serif";
}

.background {
  position: relative;
  width: 100%;
  height: 100vh;
  background: url("../public/fantasyBackground.png") no-repeat center
    center/cover;
  overflow: hidden;
}

.background::before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.3); /* Adjust the opacity as needed */
  z-index: 1;
}

.content {
  position: relative;
  z-index: 2;
  color: white; /* Ensure text is visible on the dark overlay */
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
}
