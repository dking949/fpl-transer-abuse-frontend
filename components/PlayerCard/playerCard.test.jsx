import React from 'react';
import {
  render,
} from '@testing-library/react';
import { expect, test } from '@jest/globals';
import PlayerCard from './PlayerCard';

const dummyImageUrl = 'https:resources.premierleague.com/premierleague/photos/players/110x140/p223094.png';

test('PlayerCard matches snapshot', () => {
  const component = render(<PlayerCard player_name="John" points={10} photo_url={dummyImageUrl} />);

  expect(component.container).toMatchSnapshot();
});
