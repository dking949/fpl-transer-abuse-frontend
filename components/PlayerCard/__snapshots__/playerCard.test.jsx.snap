// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerCard matches snapshot 1`] = `
<div>
  <div
    class="JoyCard-root JoyCard-variantOutlined JoyCard-colorNeutral JoyCard-sizeMd styled__CardStyled-sc-mdm4t5-0 gSCUMQ css-1hewnn1-JoyCard-root"
  >
    <div
      class="JoyCardOverflow-root JoyCardOverflow-variantPlain JoyCardOverflow-colorNeutral css-17s582v-JoyCardOverflow-root"
      data-first-child=""
    >
      <div
        class="JoyAspectRatio-root css-otcn9f-JoyAspectRatio-root"
      >
        <div
          class="JoyAspectRatio-content JoyAspectRatio-variantSoft JoyAspectRatio-colorNeutral css-hfdg0j-JoyAspectRatio-content"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27300%27%20height=%27300%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Player Pic"
              data-first-child=""
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
      </div>
    </div>
    <h2
      class="JoyTypography-root JoyTypography-h2 css-1fr557m-JoyTypography-root"
    >
      John
    </h2>
    <p
      class="JoyTypography-root JoyTypography-body2 css-1muanya-JoyTypography-root"
      data-last-child=""
    >
      Points:
       
      10
    </p>
  </div>
</div>
`;
